src/internal/testutil/cmdtest/mock_client.go

```diff
@@ -0,0 +1,625 @@
+}
+func (m *mockEnterpriseClient) Heartbeat(ctx context.Context, in *enterprise.HeartbeatRequest, opts ...grpc.CallOption) (*enterprise.HeartbeatResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*enterprise.HeartbeatResponse), args.Error(1)
+}
+func (m *mockEnterpriseClient) Deactivate(ctx context.Context, in *enterprise.DeactivateRequest, opts ...grpc.CallOption) (*enterprise.DeactivateResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*enterprise.DeactivateResponse), args.Error(1)
+}
+
+type mockLicenseClient struct {
+	mock.Mock
+}
+
+func (m *mockLicenseClient) Activate(ctx context.Context, in *license.ActivateRequest, opts ...grpc.CallOption) (*license.ActivateResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*license.ActivateResponse), args.Error(1)
+}
+func (m *mockLicenseClient) GetActivationCode(ctx context.Context, in *license.GetActivationCodeRequest, opts ...grpc.CallOption) (*license.GetActivationCodeResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*license.GetActivationCodeResponse), args.Error(1)
+}
+func (m *mockLicenseClient) DeleteAll(ctx context.Context, in *license.DeleteAllRequest, opts ...grpc.CallOption) (*license.DeleteAllResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*license.DeleteAllResponse), args.Error(1)
+}
+func (m *mockLicenseClient) AddCluster(ctx context.Context, in *license.AddClusterRequest, opts ...grpc.CallOption) (*license.AddClusterResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*license.AddClusterResponse), args.Error(1)
+}
+func (m *mockLicenseClient) DeleteCluster(ctx context.Context, in *license.DeleteClusterRequest, opts ...grpc.CallOption) (*license.DeleteClusterResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*license.DeleteClusterResponse), args.Error(1)
+}
+func (m *mockLicenseClient) ListClusters(ctx context.Context, in *license.ListClustersRequest, opts ...grpc.CallOption) (*license.ListClustersResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*license.ListClustersResponse), args.Error(1)
+}
+func (m *mockLicenseClient) UpdateCluster(ctx context.Context, in *license.UpdateClusterRequest, opts ...grpc.CallOption) (*license.UpdateClusterResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*license.UpdateClusterResponse), args.Error(1)
+}
+func (m *mockLicenseClient) Heartbeat(ctx context.Context, in *license.HeartbeatRequest, opts ...grpc.CallOption) (*license.HeartbeatResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*license.HeartbeatResponse), args.Error(1)
+}
+func (m *mockLicenseClient) ListUserClusters(ctx context.Context, in *license.ListUserClustersRequest, opts ...grpc.CallOption) (*license.ListUserClustersResponse, error) {
+	args := m.Called(in)
+	return args.Get(0).(*license.ListUserClustersResponse), args.Error(1)
+}
+
+func RunPachctlSync(t *testing.T, env cmdutil.Env, args ...string) error {
```

Comment:
don't love that this is just hiding at the bottom :P

looking at the test usage, I wonder if it would be less painful to just have a single args string that we parse